File: Solved/problem50.py
#!/usr/bin/python

def isprime(num):
    if num<=1:
        return False
    prime=0
    if num==2:
        return True
    if num%2==0:
        return False
    for i in xrange(3,int(num**0.5)+1,2):
        if num%i==0:
            prime=1
            break
    if prime==0:
        return True
    else:
        return False

limit=1000000
primes=[2]
def findnextitem(n,liste):
    return min(xrange(len(liste)), key=lambda i: abs(liste[i]-n))

for i in xrange(3,limit,2):
    if isprime(i):
        primes.append(i)

maxlenght=1
for prime in primes[-1000:]:
    if maxlenght>10:
        searchsize=findnextitem(prime/200.,primes)
    else:
        searchsize=findnextitem(prime/2.,primes)
    for i in xrange(searchsize):
        for j in xrange(searchsize-i):
            primesum=sum(primes[i:j+i])
            if prime==primesum:
                if maxlenght<j:
                    maxlenght=j
                    print "New record",prime,"with",maxlenght,"primes."
            elif prime<primesum:
                break
                

File: Solved/problem50.cpp
#include<iostream>
#include<vector>
#include<math.h>
using namespace std;
bool isPrime(int n) {
    if (n<=1) return false;
    if (n==2) return true;
    if (n%2==0) return false;
    for (int i=3;i<=sqrt(n);i+=2) if (n%i==0) return false;
    return true;
    }
    
int main() {
    int maxsteps=0;
    vector<int> primevec;
    for(int i=0;i<50000;i++) if (isPrime(i)) primevec.push_back(i);
    for (int i=0;i<primevec.size();i++) {
        int sum=0;
        int sumprimes=0;
        for(int j=i;j<primevec.size();j++) {
            sum+=primevec[j];
            sumprimes++;
            if (sum>1000000) break;
            if (sumprimes>maxsteps) {
                if (isPrime(sum)) {
                    cout << "new record " << sum << " with " << sumprimes << " primes." << endl;
                    maxsteps=sumprimes;
                    }
                }
            }
        }
    }
